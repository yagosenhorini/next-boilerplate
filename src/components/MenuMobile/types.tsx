export type IMenuMobile = {
  isActive: boolean;
  setIsActive: () => void;
  children: React.ReactNode;
};
