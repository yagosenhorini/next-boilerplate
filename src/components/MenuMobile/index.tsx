import * as S from './styled';

import { IMenuMobile } from './types';

const MenuMobile = ({ isActive, setIsActive, children }: IMenuMobile) => (
  <S.MenuWrapper data-testid="menu-list-mobile" $isActive={isActive}>
    <S.MenuList>
      <S.CloseButton type="button" onClick={setIsActive}>
        X
      </S.CloseButton>
      {children}
    </S.MenuList>
  </S.MenuWrapper>
);
export default MenuMobile;
