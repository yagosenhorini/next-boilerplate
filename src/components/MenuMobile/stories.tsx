import { Meta, StoryObj } from '@storybook/react';
import MenuMobile from '.';

export default {
  title: 'MenuMobile',
  component: MenuMobile,
  tags: ['autodocs'],
} as Meta;

export const Default: StoryObj = {};
