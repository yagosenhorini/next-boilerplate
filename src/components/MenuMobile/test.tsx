import { render, screen } from '@testing-library/react';

import MenuMobile from '.';

import StyledProvider from '../../lib/theme';

describe('<MenuMobile />', () => {
  it('should render the component', () => {
    const { container } = render(
      <StyledProvider>
        <MenuMobile isActive={true} setIsActive={jest.fn()}>
          <p>oi</p>
        </MenuMobile>
      </StyledProvider>
    );

    expect(screen.getByTestId('menu-list-mobile')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
