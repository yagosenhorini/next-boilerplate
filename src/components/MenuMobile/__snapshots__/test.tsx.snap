// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuMobile /> should render the component 1`] = `
.c0 {
  width: 100%;
  height: 100vh;
  position: fixed;
  transition: transform ease-in-out 0.7s;
  transform: translateX(0%);
  background-color: #7a8391;
}

.c1 {
  display: flex;
  list-style: none;
  height: 18.75rem;
  flex-direction: column;
  justify-content: space-between;
}

.c1 a {
  text-decoration: none;
}

.c1 li {
  font-size: 1.5rem;
  margin-left: 1.875rem;
}

.c2 {
  border: none;
  cursor: pointer;
  width: 2.5rem;
  height: 2.5rem;
  font-size: 1rem;
  color: #000;
  background-color: #E5E5E5;
  border: 0.0625rem solid #000;
}

<div
  class="c0"
  data-testid="menu-list-mobile"
>
  <ul
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      X
    </button>
    <p>
      oi
    </p>
  </ul>
</div>
`;
